### ISSUE TITLE
[HxGrid] When calling RefreshDataAsync() not UI thread, grid does not automatically rerender

### ISSUE THREAD

**Issue by TeaTime762** [havit/Havit.Blazor#102]
When I want to call RefreshDataAsync and I am not on UI thread I have to call `InvokeAsync`. That's pretty clear.
But unfortunately the grid does not update so as a workaround I call `InvokeAsync(StateHasChanged)` and it works.
Is this behaviour intended?

**Comment by hakenr** [comment]
Well, it depends...
Can you provide us with a simple repro code? ...we might come to the conclusion that it makes sense to call `StateHasChanged()` from `RefreshDataAsync()` explicitly.

cc @jirikanda

**Comment by TeaTime762** [comment]
Is this sufficient?


This method is called from the backend, whenever the UI has to update.
`
// this code works perfectly fine	
private void AlarmController_AlarmsChanged(object sender, EventArgs e)
	{
		InvokeAsync(async () => await gridComponent.RefreshDataAsync());
	         InvokeAsync(StateHasChanged);
	}
	`
	
	But I read in your documentation that there is actually no need for calling StateHasChanged() explicitly.
	Nevertheless the following code does not work, the UI does not present the latest data.
	
	`
// this code does not work
private void AlarmController_AlarmsChanged(object sender, EventArgs e)
	{
		InvokeAsync(async () => await gridComponent.RefreshDataAsync());
	}
	`

**Comment by jirikanda** [comment]
@TeaTime762 is right, the grid does not work well in this scenario. See Issue102Test.
It works well in the InfiniteScroll mode.

Internally, the RefreshDataAsync methods calls RefreshPaginationDataCoreAsync(renderOnSuccess: false) and because of the renderOnSuccess parameter value, StateHasChanged method is not called.

And we use false value with this explanation which is not valid here (not a typical use case).
``` 
// We don't auto-render after this operation because in the typical use case, the
// host component calls this from one of its lifecycle methods, and will naturally
// re-render afterwards anyway. It's not desirable to re-render twice.
```

To make it work:
* developer should call StateHasChange on his own
* or we should call StateHasChange always (in typical scenario the StateHasChanged will be fired twice - first time from this RefreshDataAsync/RefreshPaginationDataCoreAsync, second time it will be propagated from the parent component re-render).
* or we could add argument to render to the RefreshDataAsync method (or overload the method), and let the developer to choose. But there is a scenarion with Virtualize component where the parameter is useless.
* or ?

@hakenr

**Comment by hakenr** [comment]
Fixed in v2.3.0

**Comment by TeaTime762** [comment]
Awesome guys! Thank you very much!
